.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,105 @@
+name: Benchmark
+
+# on:
+#   pull_request:
+#     types: [labeled]
+#   workflow_dispatch:
+on:
```

Comment:
Seems excessive to run this on every pull request.

Running when a particular label is applied seems nice, since we can turn it on when necessary.

I think we could also run it when merging to `main`, and in the future configure it to automatically open an issue if there's a regression.

Maybe @andersy005 and @keewis have ideas here.

Regardless, I think we could merge the rest and debate the CI setup in a future PR.

Reply:
> Seems excessive to run this on every pull request.

I concur... Looks like the current benchmarks are taking ~ 30 minutes to run, and it's probably safe to assume that this runtime will increase once we expand the benchmarks. So, I think we should limit the benchmark CI to labeled PRs and/or to the commits to the main branch. 

> I think we could also run it when merging to main, and in the future configure it to automatically open an issue if there's a regression.

> Regardless, I think we could merge the rest and debate the CI setup in a future PR.

👍🏽  👍🏽

Reply:
My take on this is that it will be maintained much better if it's run more often. 
Setting up asv isn't that easy either so it might get a lot more interest if it just runs.

This PRs benchmarks takes 30 minutes now. The version on main takes _240+_ minutes with a lot of errors and warnings in the report making it very hard to read. 
Were those errors always there it or have they appeared over time? Considering the slowness I'm not so sure there are that many that actually managed to run the benchmarks and had the patience to wait until it finished to check for these things.

Reply:
Now it triggers only when the `run-benchmark` label is set otherwise it is skipped. 

With scikit-image version you have to manually remove and add back the label for the job to trigger everytime you want it to run, I found that behaviour annoying when I was frequently testing the benchmarks with the CI.